src/topology/alexandroff.lean

```diff
@@ -0,0 +1,369 @@
+
+/-- An open set in `alexandroff X` constructed from a closed compact set in `X` -/
+def opens_of_compl (s : set X) (h₁ : is_closed s) (h₂ : is_compact s) :
+  topological_space.opens (alexandroff X) :=
+⟨(coe '' s)ᶜ, is_open_compl_image_coe.2 ⟨h₁, h₂⟩⟩
+
+lemma infty_mem_opens_of_compl {s : set X} (h₁ : is_closed s) (h₂ : is_compact s) :
+  ∞ ∈ opens_of_compl s h₁ h₂ :=
+mem_compl infty_not_mem_image_coe
+
+@[continuity] lemma continuous_coe : continuous (coe : X → alexandroff X) :=
+continuous_def.mpr (λ s hs, hs.right)
+
+lemma is_open_map_coe  : is_open_map (coe : X → alexandroff X) :=
+λ s, is_open_image_coe.2
+
+lemma open_embedding_coe : open_embedding (coe : X → alexandroff X) :=
+open_embedding_of_continuous_injective_open continuous_coe coe_injective is_open_map_coe
+
+lemma is_open_range_coe : is_open (range (coe : X → alexandroff X)) :=
+open_embedding_coe.open_range
+
+lemma is_closed_infty : is_closed ({∞} : set (alexandroff X)) :=
+by { rw [← compl_range_coe, is_closed_compl_iff], exact is_open_range_coe }
+
+lemma nhds_coe_eq (x : X) : 𝓝 ↑x = map (coe : X → alexandroff X) (𝓝 x) :=
+(open_embedding_coe.map_nhds_eq x).symm
+
+lemma nhds_within_coe_image (s : set X) (x : X) :
+  𝓝[coe '' s] (x : alexandroff X) = map coe (𝓝[s] x) :=
+by rw [nhds_within, nhds_within, map_inf coe_injective, nhds_coe_eq, map_principal]
+
+lemma comap_coe_nhds (x : X) : comap (coe : X → alexandroff X) (𝓝 x) = 𝓝 x :=
+(open_embedding_coe.to_inducing.nhds_eq_comap x).symm
+
+lemma nhds_infty_eq : 𝓝 (∞ : alexandroff X) = map coe (coclosed_compact X) ⊔ pure ∞ :=
+begin
+  refine (nhds_basis_opens ∞).ext ((has_basis_coclosed_compact.map _).sup_pure _) _ _,
+  { rintro s ⟨hs, hso⟩,
+    refine ⟨_, (is_open_iff_of_mem hs).mp hso, _⟩,
+    simp [insert_subset, hs] },
+  { rintro s ⟨h₁, h₂⟩,
+    refine ⟨_, ⟨mem_compl infty_not_mem_image_coe, is_open_compl_image_coe.2 ⟨h₁, h₂⟩⟩, _⟩,
+    rw compl_image_coe }
+end
+
+lemma has_basis_nhds_infty :
+  (𝓝 (∞ : alexandroff X)).has_basis (λ s : set X, is_closed s ∧ is_compact s)
+    (λ s, coe '' sᶜ ∪ {∞}) :=
+begin
+  rw nhds_infty_eq,
+  exact (has_basis_coclosed_compact.map _).sup_pure _
+end
+
+@[simp] lemma comap_coe_nhds_infty : comap (coe : X → alexandroff X) (𝓝 ∞) = coclosed_compact X :=
+by simp [nhds_infty_eq, comap_sup, comap_map coe_injective]
+
+lemma le_nhds_infty {f : filter (alexandroff X)} :
+  f ≤ 𝓝 ∞ ↔ ∀ s : set X, is_closed s → is_compact s → coe '' sᶜ ∪ {∞} ∈ f :=
+by simp only [has_basis_nhds_infty.ge_iff, and_imp]
+
+lemma ultrafilter_le_nhds_infty {f : ultrafilter (alexandroff X)} :
+  (f : filter (alexandroff X)) ≤ 𝓝 ∞ ↔ ∀ s : set X, is_closed s → is_compact s → coe '' s ∉ f :=
+by simp only [le_nhds_infty, ← compl_image_coe, ultrafilter.mem_coe,
+  ultrafilter.compl_mem_iff_not_mem]
+
+lemma tendsto_nhds_infty {α : Type*} {f : alexandroff X → α} {l : filter α} :
+  tendsto f (𝓝 ∞) l ↔
+    ∀ s ∈ l, f ∞ ∈ s ∧ ∃ t : set X, is_closed t ∧ is_compact t ∧ maps_to (f ∘ coe) tᶜ s :=
```

Comment:
It would be nice to have versions `tendsto_nhds_infty'` and `continuous_at_infty'` for the more abstract versions using filters, which might even be more useful.  If I understand correctly these should be
```
tendsto f (𝓝 ∞) l ↔ tendsto f (pure ∞) l ∧ tendsto (f ∘ coe) (coclosed_compact X) l
continuous_at f ∞ ↔ tendsto (f ∘ coe) (coclosed_compact X) (𝓝 (f ∞))
```
is that right?

Reply:
Done. BTW, could you please write ```` ```lean ... ``` ```` instead of just ```` ``` ... ``` ```` to get syntax highlighting?